src/hydrusvideodeduplicator/dedup.py

```diff
@@ -133,6 +134,25 @@ def _retrieve_video_hashes(
         )["hashes"]
         return all_video_hashes
 
+    def _fetch_and_hash_file(self, video_hash: str) -> dict | None:
+        try:
+            video_response = self.client.get_file(hash_=video_hash)
+        except hydrus_api.HydrusAPIException:
+            rprint("[red] Failed to get video from Hydrus.")
+            self.hydlog.error("Error getting video from Hydrus.")
+            return None
+
+        # Calculate perceptual_hash
+        try:
+            perceptual_hash = self._calculate_perceptual_hash(video_response.content)
+        except Exception as exc:
+            rprint("[red] Failed to calculate a perceptual hash.")
+            self.hydlog.exception(exc)
+            self.hydlog.error(f"Errored file hash: {video_hash}")
+            return None
+        else:
+            return {"video_hash": video_hash, "perceptual_hash": perceptual_hash}
```

Comment:
return as tuple instead of dict

Reply:
@ianwal Any particular reason why? Accessing tuples means using numbers to access everything, and I generally find that makes any code using returned tuples less clear.

Reply:
You can use a namedtuple if you want even clearer access. Dict's are just overkill here.

Reply:
@ianwal Oho, I didn't know about namedtuples before this! Nice 👍🏼